### ISSUE TITLE
Is your work based on unsupervised learning?

### ISSUE THREAD

**Issue by estelle1722** [jantic/DeOldify#6]
Hi!
Your work is great meaningful.I wonder whether it is unsupervised and am interested in the reasons why you don't choose wgan or wgan-gp finally?

thanks.

**Comment by estelle1722** [comment]
Oh,I'm sorry.Regarding 
as 
being unsupervised,I want to know whether groundtruth is used in your work.

**Comment by jantic** [comment]
So what it does is strictly supervised learning when it comes to training the generator on perceptual loss- that is, the loss that encourages the generator to replicate the black and white input image (the target is the color version of the same image).  Now the reason why I say I'm fuzzy on definitions here is because the critic portion is looking at real and fake versions and being asked to assign a score to each for "realism", but it's not actually being told that the fake image and real image should match per se.  So all the critic can tell the generator then is that "that's not realistic" and by how much according to a numerical score.  But the end result is that taken together this combination drives the generator to create vividly colorful transformations of the input black & white images.

So mostly yes on "ground truth is used" but it's complicated I guess.... I might just be confused myself on definitions though honestly...

**Comment by estelle1722** [comment]
Taking classification problem as an example,given an image of cat,its groundtruth or label is cat.In your work,what I say groundtruth is correspondingly correct or standard  transformations  given input black images.From your reply,I guess there is no corresponding groundtruth given an input black image.
thanks.

**Comment by estelle1722** [comment]
Taking classification problem as an example,given an image of cat,its groundtruth or label is cat.In your work,what I say groundtruth is the color version of the same image given an input black image.

**Comment by jantic** [comment]
Yes that's what I meant when I said "that is, the loss that encourages the generator to replicate the black and white input image (the target is the color version of the same image)."

It's just Imagenet photos being converted to gray scale, then the neural net's job is to convert it back to color. But I say it's complicated because that's not the complete picture (see above).
